pkg/resource/certificate_authority_activation/hooks.go

```diff
@@ -192,3 +192,39 @@ func (rm *resourceManager) customUpdateCertificateAuthorityActivation(
 	rm.setStatusDefaults(ko)
 	return &resource{ko}, nil
 }
+
+func (rm *resourceManager) customDeleteCertificateAuthorityActivation(
+	ctx context.Context,
+	r *resource,
+) (latest *resource, err error) {
+
+	input := &svcsdk.DescribeCertificateAuthorityInput{}
+	input.CertificateAuthorityArn = r.ko.Spec.CertificateAuthorityARN
+
+	var resp *svcsdk.DescribeCertificateAuthorityOutput
+	resp, err = rm.sdkapi.DescribeCertificateAuthorityWithContext(ctx, input)
+	rm.metrics.RecordAPICall("READ_ONE", "DescribeCertificateAuthority", err)
+	if err != nil {
+		return nil, err
+	}
+
+	if *resp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusDeleted && *resp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusDisabled {
```

Comment:
You need to check that the status is not nil - otherwise the controller might crash if for some reason the status wasn't populated